packages/flutter_service_worker/README.md

```diff
@@ -0,0 +1,67 @@
+## Flutter Service Worker
+
+A collection of utility APIs for interacting with the Flutter service worker.
+
+
+### Setup
+
+The `init` method should be called in main to bootstrap the service worker API. This is safe
+to call on non-web platforms.
+
+```dart
+void main() {
+  serviceWorkerApi.init();
+  runApp(MyApp());
+}
+```
+
+
+### Detecting a new version
+
+A service worker will cache the old application until the new application is downloaded and ready. To be notified when this occurs, listen to the `newVersionReady` future. You can then use `skipWaiting()` to
+force-load the new version.
+
+```dart
+
+serviceWorkerApi.newVersionReady.whenComplete(() {
+  showNewVersionDialog().then((bool yes) {
```

Comment:
Where is this function?

Reply:
`showNewVersionDialog()` is an example function that you might implement with `showDialog` or some other flutter functionality. I will fill out these examples more so they reference only working flutter code